src/lib/sync-engine.ts

```
@@ -0,0 +1,718 @@
+      do {
+        const req: SyncRequest = {
+          lastPulledAt,
+          schemaVersion: '2025-08-23.v1',
+          tables: SYNC_TABLES,
+          cursor,
+        };
+        const resp = await pullChangesOnce(req);
+        await applyServerChanges(resp);
+        cursor = resp.hasMore ? resp.nextCursor : undefined;
+        if (!resp.hasMore) break;
+      } while (true);
```

Biome (2.1.2)

[error] Unexpected constant condition.



(lint/correctness/noConstantCondition)